src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubAppCredentials.java

```diff
@@ -72,27 +77,55 @@ public Secret getPrivateKey() {
         return privateKey;
     }
 
+    /**
+     * Owner of this installation, i.e. a user or organisation,
+     * used to differeniate app installations when the app is installed to multiple organisations / users.
+     *
+     * If this is null then call listInstallations and if there's only one in the list then use that installation.
+     *
+     * @return the owner of the organisation or null.
+     */
+    @CheckForNull
+    public String getOwner() {
+        return owner;
+    }
+
+    @DataBoundSetter
+    public void setOwner(String owner) {
+        this.owner = owner;
+    }
+
     @SuppressWarnings("deprecation") // preview features are required for GitHub app integration, GitHub api adds deprecated to all preview methods
-    static String generateAppInstallationToken(String appId, String appPrivateKey, String apiUrl) {
+    static String generateAppInstallationToken(String appId, String appPrivateKey, String apiUrl, String owner) {
         try {
             String jwtToken = createJWT(appId, appPrivateKey);
             GitHub gitHubApp = new GitHubBuilder().withEndpoint(apiUrl).withJwtToken(jwtToken).build();
 
             GHApp app = gitHubApp.getApp();
 
             List<GHAppInstallation> appInstallations = app.listInstallations().asList();
-            if (!appInstallations.isEmpty()) {
-                GHAppInstallation appInstallation = appInstallations.get(0);
-                GHAppInstallationToken appInstallationToken = appInstallation
-                        .createToken(appInstallation.getPermissions())
-                        .create();
-
-                return appInstallationToken.getToken();
+            if (appInstallations.isEmpty()) {
+                throw new IllegalArgumentException(String.format(ERROR_NOT_INSTALLED, appId));
             }
+            GHAppInstallation appInstallation;
+            if (appInstallations.size() == 1) {
+                appInstallation = appInstallations.get(0);
```

Comment:
Not verifying the owner?

Reply:
I've made an assumption that most installations only work with one org, so I've made it an advanced field and by default if and only if there's one installation the plugin will just pick that.

Reply:
I get that, I am just saying that in case you did specify an owner, and there is exactly one installation, we are not verifying that the installation is in fact from that owner. Which feels wrong.